projects/demo/src/app/demo/pages/select-page/select-page.component.html

```diff
@@ -2,74 +2,281 @@
+        </ul>
+
+        <div *ngIf="selectedTab === 'standalone'">
+            <pa-select
+                label="Html Options, required"
+                placeholder="Placeholder"
+                required
+                [value]="selectedValue1"
+                [disabled]="disabledState"
+                [readonly]="readOnlyState"
+                [hasFocus]="firstFocused"
+                [errorMessages]="errorMessages"
+                (valueChange)="valueChange = $event"
+                (expanded)="expandedEvent = $event"
+            >
+                <pa-option-header>Users</pa-option-header>
+                <pa-option value="user1">User 1</pa-option>
+                <pa-option value="user2">User 2</pa-option>
+                <pa-option value="user3">User 3</pa-option>
+                <pa-option-header>Groups</pa-option-header>
+                <pa-option value="group1" disabled>Group 1</pa-option>
+                <pa-option value="group2">Group 2</pa-option>
+                <pa-option value="group3">Group 3</pa-option>
+            </pa-select>
+
+            <pa-select
+                label="Options passed as parameter"
+                placeholder="Placeholder"
+                [options]="options"
+                [disabled]="disabledState"
+                [readonly]="readOnlyState"
+                (valueChange)="valueChange = $event"
+                (expanded)="expandedEvent = $event"
+            ></pa-select>
+
+            <pa-select
+                label="Not preselected, free text, pattern validation"
+                placeholder="Placeholder"
+                name="notPreselected"
+                [pattern]="pattern"
+                [disabled]="disabledState"
+                [readonly]="readOnlyState"
+                [errorMessages]="errorMessages"
+                [suggestionMode]="true"
+                (valueChange)="valueChange = $event"
+                (expanded)="expandedEvent = $event"
+            >
+                <pa-option-header>Users</pa-option-header>
+                <pa-option value="user1">User 1</pa-option>
+                <pa-option value="user2">User 2</pa-option>
+                <pa-option value="user3">User 3</pa-option>
+                <pa-option-header>Groups</pa-option-header>
+                <pa-option value="group1" disabled>Group 1</pa-option>
+                <pa-option value="group2">Group 2</pa-option>
+                <pa-option value="group3">Group 3</pa-option>
+            </pa-select>
+        </div>
+
+        <div *ngIf="selectedTab === 'ngModel'">
+            <pa-select
+                label="Html Options, required"
+                placeholder="Placeholder"
+                required
+                [(ngModel)]="selectedValue1"
+                [disabled]="disabledState"
+                [readonly]="readOnlyState"
+                [hasFocus]="firstFocused"
+                [errorMessages]="errorMessages"
+                (valueChange)="valueChange = $event"
```

Comment:
ngModel and formControls change is handled by angular... the only reason to have a valueChange is the standalone mode... 
I could prevent valueChange to emit in these cases but I thought that was not necessary since it's working fine the way it is.

The reason why valueChange is emiting in other modes than standalone is that we are managing the value programatically all the time whereas in input component we fully delegate this to angular for ngModel and formControl modes

Reply:
yes, sorry I wasn't clear: I didn't mean you should prevent the event to be triggered, I meant you should not include it in the code example / documentation when it doesn't bring any value. All to avoid confusion for people who will use our components and go through the documentation and documentation page to understand what they should do.